lib/Doctrine/Migrations/MigrationTableManager.php

```diff
@@ -0,0 +1,224 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Migrations;
+
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\Platforms\AbstractPlatform;
+use Doctrine\DBAL\Schema\AbstractSchemaManager;
+use Doctrine\DBAL\Schema\Column;
+use Doctrine\DBAL\Schema\Schema;
+use Doctrine\DBAL\Schema\Table;
+use Doctrine\DBAL\Types\Type;
+use Doctrine\Migrations\Configuration\Configuration;
+use function array_keys;
+use function in_array;
+
+/**
+ * @internal
+ */
+class MigrationTableManager
+{
+    private const MIGRATION_COLUMN_TYPE = 'string';
+
+    /** @var Configuration */
+    private $configuration;
+
+    /** @var AbstractSchemaManager */
+    private $schemaManager;
+
+    /** @var Connection */
+    private $connection;
+
+    /** @var AbstractPlatform */
+    private $platform;
+
+    /** @var string */
+    private $tableName;
```

Comment:
I think it might be time to have a class to represent the migration table, which encapsulates: name, column name (needs a better name), column length (better name as well?), and executed at column name.

Reply:
Good idea. I will make that change.